WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -173,6 +246,60 @@ private extension PointOfSaleDashboardViewModel {
             }
         }.assign(to: &$showsCardReaderSheet)
     }
+
+    func observeProductsInCartForRemoteOrderSyncing() {
```

Comment:
My understanding is that this method is used to ensure that the remote order is kept up-to-date with the local cart's contents, right? I think there are too many things happening inside the function and I'm not quite sure to follow every step. Can we offload some of the logic to different functions and add comments to what's happening? Eg: mapCartProducts(), `makeOrder()`, `syncOrder()` ...